__tests__/server/middleware/pwa/manifest.spec.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  webmanifestMiddleware,
+  configureWebManifest,
+  getWebManifestEnabled,
+  getWebManifest,
+  setWebManifest,
+  resetWebManifest,
+} from '../../../../src/server/middleware/pwa/manifest';
+
+describe('webmanifest middleware', () => {
+  test('exports functions as default', () => {
+    expect.assertions(6);
+    expect(webmanifestMiddleware).toBeInstanceOf(Function);
+    expect(configureWebManifest).toBeInstanceOf(Function);
+    expect(getWebManifestEnabled).toBeInstanceOf(Function);
+    expect(getWebManifest).toBeInstanceOf(Function);
+    expect(setWebManifest).toBeInstanceOf(Function);
+    expect(resetWebManifest).toBeInstanceOf(Function);
+  });
+
+  test('gets a valid webmanifest and gets enabled value', () => {
+    expect.assertions(2);
+    expect(getWebManifestEnabled()).toEqual(false);
+    expect(getWebManifest()).toBe(null);
+  });
+
+  test('resets the webmanifest', () => {
+    expect.assertions(1);
+    expect(resetWebManifest()).toMatchObject({
+      name: expect.any(String),
+      short_name: expect.any(String),
+    });
+  });
+
+  test('sets the webmanifest value', () => {
+    expect.assertions(3);
+    expect(setWebManifest({ name: 'One App Test' })).toMatchObject({
+      name: 'One App Test',
+      short_name: expect.any(String),
+    });
+    expect(getWebManifestEnabled()).toEqual(false);
+    expect(getWebManifest()).toMatchObject({
+      name: 'One App Test',
+      short_name: expect.any(String),
+    });
+  });
+
+  test('configuring the webmanifest without params does nothing', () => {
+    expect.assertions(3);
+    expect(configureWebManifest()).toBeUndefined();
+    expect(getWebManifestEnabled()).toEqual(false);
+    expect(getWebManifest()).toEqual(null);
+  });
+
+  test('configures the webmanifest and enables it', () => {
+    expect.assertions(3);
+    expect(configureWebManifest({ enabled: true, manifest: { short_name: 'one-app-test' } })).toBeUndefined();
+    expect(getWebManifestEnabled()).toEqual(true);
+    expect(getWebManifest()).toMatchObject({
+      name: expect.any(String),
+      short_name: expect.any(String),
```

Comment:
why wouldn't this be `"one-app-test"`?

Reply:
No reason in particular, I figured checking the type would suffice. I will update this in future PR, removed manifest here for now.